pkg/render/windows.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package render
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/components"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+)
+
+func Windows(
+	cr *operatorv1.InstallationSpec,
+	hasSupportedNodes bool,
```

Comment:
How are we going to calculate this variable? Is this just "Does the cluster have Windows nodes"? Or is it "Does the cluster have windows nodes that need to be upgraded"? Or something else?

Reply:
The variable was used when we were still rendering the upgrade daemonset only if there were windows nodes to upgrade. I removed it since in an earlier review we had agreed that it probably wasn't worth the extra logic.